Validate reserved names (#2434)
getporter/porter: docs/content/reference/linter.md

To fix the problem indicated by the exec-100 warning, you can try replacing any embedded Bash scripts in your exec mixins with scripts.

You can find more information about best practices for using the exec mixin on the Porter website at the following URL: https://getporter.org/best-practices/exec-mixin/#use-scripts.

## porter-100

> The porter-100 warning is a message generated by the porter lint command when it detects when a  parameter name has one of the reserved prefixes `porter-` and `porter_`.

Reviewer: ```suggestion
The porter-100 error is a message generated by the porter lint command when it detects when a  parameter name has one of the reserved prefixes `porter-` and `porter_`.
```

> The porter-100 warning suggests that a pararmeter has been created which uses one of the reserved prefixes.

Reviewer: ```suggestion
The porter-100 error suggests that a parameter has been created which uses one of the reserved prefixes.
```

Using a reserved prefix can be problematic as it can overwrite a predefined parameter resulting in undeterministic behaviour.

> To fix the problem indicated by the porter-100 warning, you can try replacing the prefix of any newly defined parameters.

Reviewer: ```suggestion
To fix the problem indicated by the porter-100 error, you should replace the prefix of any newly defined parameters to not start with "porter".
```
